packages/office-ui-fabric-react/src/components/Sticky/Sticky.tsx

```diff
@@ -105,20 +108,33 @@ export class Sticky extends BaseComponent<IStickyProps, IStickyState> {
     if (prevState.isStickyTop !== this.state.isStickyTop || prevState.isStickyBottom !== this.state.isStickyBottom) {
       scrollablePane.updateStickyRefHeights();
     }
+
+    if (this.state.distanceFromTop !== prevState.distanceFromTop) {
+      scrollablePane.sortSticky(this, true);
+    }
   }
 
   public shouldComponentUpdate(nextProps: IStickyProps, nextState: IStickyState): boolean {
     if (!this.context.scrollablePane) {
       return true;
     }
 
-    const { isStickyTop, isStickyBottom } = this.state;
-    return (
-      isStickyTop !== nextState.isStickyTop ||
-      isStickyBottom !== nextState.isStickyBottom ||
-      this.props.stickyPosition !== nextProps.stickyPosition ||
-      this.props.children !== nextProps.children
-    );
+    const { isStickyTop, isStickyBottom, distanceFromTop } = this.state;
+
+    if (this._nonStickyContent && this._nonStickyContent.current && this._placeHolder && this._placeHolder.current) {
+      const nonStickyContentHeight = this._nonStickyContent.current.offsetHeight;
+
+      return (
+        isStickyTop !== nextState.isStickyTop ||
+        isStickyBottom !== nextState.isStickyBottom ||
+        distanceFromTop !== nextState.distanceFromTop ||
+        this.props.stickyPosition !== nextProps.stickyPosition ||
+        this.props.children !== nextProps.children ||
+        nonStickyContentHeight !== this._placeHolder.current.offsetHeight
```

Comment:
What about the stickyContentTop/stickyContentBottom container offsetHeight? In scenarios where the nonStickyContent offsetHeight has changed, the stickyContentTop offsetHeight will not be updated with this code.

I don't have a scenario in mind where this may cause issues. But since we are already maintaining a stickyContentTop container with a div which has the offsetHeight of nonStickyContent, I am sure you had reasons to do that in the first place.

Reply:
`stickyContentTop`'s height is set by `nonStickyContent` when it's moved in the DOM when it's considered Sticky.   If this height changes, then `ScrollablePane` will see this change with the `MutationObserver` and update the heights accordingly (~line 102 in `ScrollablePane.base.tsx`)

Reply:
To clarify, I meant the _div_ inside _stickyContentTop_ and _stickyContentBottom_ which has been styled (using __getStickyPlaceholderHeight_) to a height of the _nonStickyContent_, when the component is **not sticky**.